### PR TITLE
Clean up detail controllers and add unit tests

### PR DESCRIPTION
**Pull request by Sheikah45**

### CODE REVIEW

**Review by Brutus5000 — COMMENTED** [review]
Please don't name a PR add unit tests and then add changes in the main application.
I didn't intend to review unit tests.

File: src/main/java/com/faforever/client/mod/Mod.java [hunk 1]
```diff
@@ -19,7 +18,7 @@
   private final StringProperty author;
   private final ObjectProperty<OffsetDateTime> createTime;
   private final ObjectProperty<OffsetDateTime> updateTime;
-  private final ObjectProperty<Player> uploader;
+  private final StringProperty uploader;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/mod/Mod.java:21]: Why move to String?

File: src/main/resources/i18n/messages.properties [hunk 2]
```diff
@@ -890,5 +890,7 @@ games.create.enforceRating=Enforce rating
 review.currentVersion=Current Version
 review.version=Version {0}
 review.date={0} on {1}
+review.delete.error=This review could not be deleted
+review.save.error=This review could not be deleted
```
Comment (Brutus5000) [inline src/main/resources/i18n/messages.properties]: could not be saved

File: src/main/java/com/faforever/client/mod/Mod.java (hunk 1 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/mod/Mod.java:21]: I moved to string for consistency with author and also since only the username is needed

**Pull request merged**